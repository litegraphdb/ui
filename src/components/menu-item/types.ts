export interface MenuItemProps {
  key: string;
  icon?: React.ReactNode;
  label?: string;
  path?: string;
  children?: MenuItemProps[];
  props?: any;
}
