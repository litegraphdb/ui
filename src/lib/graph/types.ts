export interface NodeData {
  id: string;
  label: string;
  type: string;
  x: number;
  y: number;
  vx: number;
  vy: number;
  isDragging?: boolean;
}

export interface EdgeData {
  id: string;
  source: string;
  target: string;
  cost: number;
  data: string;
  sourceX: number;
  sourceY: number;
  targetX: number;
  targetY: number;
  label?: string;
}

export type HoveredElement =
  | {
      type: 'edge';
      data: EdgeData;
    }
  | {
      type: 'node';
      data: NodeData;
    }
  | null;

export interface Point {
  x: number;
  y: number;
}
