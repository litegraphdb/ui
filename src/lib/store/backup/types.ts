import { BackupMetaData } from 'litegraphdb/dist/types/types';

export type BackupType = BackupMetaData;
