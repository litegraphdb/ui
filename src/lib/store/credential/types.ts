import { CredentialMetadata } from 'litegraphdb/dist/types/types';

export type CredentialType = CredentialMetadata;
