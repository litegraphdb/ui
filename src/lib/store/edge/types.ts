import Edge from 'litegraphdb/types/models/Edge';

export type EdgeGroupWithGraph = {
  [GraphGUID: string]: EdgeType[];
};

export type EdgeType = Edge & {
  Distance?: number;
  Score?: number;
  FromName?: string;
  ToName?: string;
};
