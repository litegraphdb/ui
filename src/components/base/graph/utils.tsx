import { NodeData } from '@/lib/graph/types';
import { defaultNodeColor, nodeLightColorByType, universalNodeColors } from './constant';

const maxLegends = 16;

export const getLegendsForNodes = (nodes: NodeData[]) => {
  const priorityLegends = new Map<string, { legend: string; color: string }>();
  const customLegends = new Map<string, { legend: string; color: string }>();

  // Separate legends into priority (nodeLightColorByType) and custom
  nodes.forEach((node) => {
    if (Object.keys(nodeLightColorByType).includes(node.type)) {
      priorityLegends.set(node.type, { legend: node.type, color: nodeLightColorByType[node.type] });
    } else {
      // Only add if not already present to avoid changing colors for same legend
      if (!customLegends.has(node.type)) {
        customLegends.set(node.type, {
          legend: node.type,
          color: universalNodeColors[customLegends.size % universalNodeColors.length],
        });
      }
    }
  });

  const priorityLegendsSize = priorityLegends.size;
  // If there are nodes with unknown types, add "Unknown" to priority legends
  const isMoreThanMaxLegends = customLegends.size + priorityLegendsSize > maxLegends;

  // Convert to arrays for easier manipulation
  const priorityArray = Array.from(priorityLegends.values());
  const customArray = Array.from(customLegends.values()).slice(
    0,
    maxLegends - (priorityArray.length + (isMoreThanMaxLegends ? 1 : 0))
  );

  // Combine priority legends first, then custom legends
  const legends = [
    ...priorityArray,
    ...customArray,
    ...(isMoreThanMaxLegends ? [{ legend: 'Unknown', color: defaultNodeColor }] : []),
  ];
  const mapLegends: Record<string, { legend: string; color: string }> = {};
  legends.forEach((legend) => {
    mapLegends[legend.legend] = legend;
  });
  return mapLegends;
};
