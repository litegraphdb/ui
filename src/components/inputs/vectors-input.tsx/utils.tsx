export const convertVectorsToAPIRecord = (
  vectors?: Array<{ Model: string; Dimensionality: number; Content: string; Vectors: number[] }>
) => {
  return (
    vectors?.map((vector: any) => ({
      ...vector,
      Dimensionality: Number(vector.Dimensionality),
      Vectors: vector.Vectors || [],
    })) || []
  );
};
