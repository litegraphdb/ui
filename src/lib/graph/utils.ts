import type { NodeData, Point } from './types';

export function isPointInNode(point: Point, node: NodeData, radius: number = 10): boolean {
  const dx = point.x - node.x;
  const dy = point.y - node.y;
  return dx * dx + dy * dy <= radius * radius;
}

export function getNodeAtPosition(nodes: NodeData[], point: Point): NodeData | null {
  return nodes.find((node) => isPointInNode(point, node)) || null;
}

export function transformToOptions<T extends { GUID: string; name?: string; Name?: string }>(
  data?: T[] | null,
  labelField: keyof T = 'name' // Field to use for label in options
) {
  return data
    ? data?.map((item) => ({
        value: item.GUID,
        label: (item[labelField] as string) || item.Name || item.GUID,
      }))
    : [];
}
