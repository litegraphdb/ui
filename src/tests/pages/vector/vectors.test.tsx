import '@testing-library/jest-dom';
import React from 'react';
import { screen, fireEvent, waitFor } from '@testing-library/react';
import VectorPage from '@/app/dashboard/[tenantId]/vectors/page';
import { createMockInitialState } from '../../store/mockStore';
import { mockVectorData, mockTenantGUID } from '../mockData';
import { setupServer } from 'msw/node';
import { handlers } from './handler';
import { commonHandlers } from '@/tests/handler';
import { renderWithRedux } from '@/tests/store/utils';
import AddEditVector from '@/page/vectors/components/AddEditVector';
import DeleteVector from '@/page/vectors/components/DeleteVector';
import { mockGraphGUID } from '../mockData';
import { VectorType } from '@/types/types';
import { http, HttpResponse } from 'msw';
import { mockEndpoint } from '@/tests/config';

// Mock react-hot-toast
jest.mock('react-hot-toast', () => ({
  success: jest.fn(),
  error: jest.fn(),
}));

const server = setupServer(...handlers, ...commonHandlers);

describe('Vectors Page', () => {
  beforeAll(() => server.listen());
  afterEach(() => {
    server.resetHandlers();
  });
  afterAll(() => server.close());

  it('renders the vectors page', () => {
    const initialState = createMockInitialState();
    const { container } = renderWithRedux(<VectorPage />, initialState, undefined, true);

    expect(screen.getByText(/vectors/i)).toBeVisible();
    expect(screen.getByRole('button', { name: /create vector/i })).toBeVisible();
    expect(container).toMatchSnapshot('initial table state');
  });

  it('should create a vector and should be visible in the table', async () => {
    const initialState = createMockInitialState();
    const { container } = renderWithRedux(<VectorPage />, initialState, undefined, true);

    const createButton = screen.getByRole('button', { name: /create vector/i });
    expect(createButton).toBeVisible();
    expect(createButton).toMatchSnapshot();
  });

  it('should create a vector and should be visible in the table', async () => {
    const initialState = createMockInitialState();
    const { container } = renderWithRedux(
      <AddEditVector
        isAddEditVectorVisible={true}
        setIsAddEditVectorVisible={() => {}}
        vector={null}
        selectedGraph={mockGraphGUID}
      />,
      initialState,
      undefined,
      true
    );

    const modal = await screen.findByTestId('add-edit-vector-modal');
    expect(modal).toBeInTheDocument();

    await waitFor(
      () => {
        expect(screen.queryByText('Loading...')).not.toBeInTheDocument();
      },
      { timeout: 10000 }
    );

    const modelInput = screen.getByPlaceholderText(/enter model/i);
    fireEvent.change(modelInput, { target: { value: mockVectorData[0].Model } });

    const dimensionalityInput = screen.getByPlaceholderText(/enter dimensionality/i);
    fireEvent.change(dimensionalityInput, { target: { value: mockVectorData[0].Dimensionality } });

    const contentInput = screen.getByPlaceholderText(/enter content/i);
    fireEvent.change(contentInput, { target: { value: mockVectorData[0].Content } });

    const submitButton = screen.getByRole('button', { name: /ok/i });
    fireEvent.click(submitButton);

    const finalTable = container.querySelector('.ant-table');
    expect(finalTable).toMatchSnapshot('final table state');
  });

  it('should update vector successfully', async () => {
    const initialState = createMockInitialState();
    const { container } = renderWithRedux(
      <AddEditVector
        isAddEditVectorVisible={true}
        setIsAddEditVectorVisible={() => {}}
        vector={null}
        selectedGraph={mockGraphGUID}
      />,
      initialState,
      undefined,
      true
    );

    const modal = await screen.findByTestId('add-edit-vector-modal');
    expect(modal).toBeInTheDocument();

    await waitFor(
      () => {
        expect(screen.queryByText('Loading...')).not.toBeInTheDocument();
      },
      { timeout: 10000 }
    );

    const modelInput = screen.getByPlaceholderText(/enter model/i);
    fireEvent.change(modelInput, { target: { value: mockVectorData[0].Model } });

    const submitButton = screen.getByRole('button', { name: /ok/i });
    fireEvent.click(submitButton);

    const finalTable = container.querySelector('.ant-table');
    expect(finalTable).toMatchSnapshot('final table state');
  });

  it('should delete vector successfully', async () => {
    const initialState = createMockInitialState();
    const { container } = renderWithRedux(
      <DeleteVector
        title="Delete Vector"
        paragraphText="Are you sure you want to delete this vector?"
        isDeleteModelVisible={true}
        setIsDeleteModelVisible={() => {}}
        selectedVector={null}
        setSelectedVector={() => {}}
        onVectorDeleted={() => Promise.resolve()}
      />,
      initialState,
      undefined,
      true
    );

    const modal = await screen.findByTestId('delete-vector-modal');
    expect(modal).toBeInTheDocument();

    await waitFor(
      () => {
        expect(screen.queryByText('Loading...')).not.toBeInTheDocument();
      },
      { timeout: 10000 }
    );

    const confirmButton = screen.getByRole('button', { name: /delete/i });
    fireEvent.click(confirmButton);

    const finalTable = container.querySelector('.ant-table');
    expect(finalTable).toMatchSnapshot('final table state');
  });
});

describe('AddEditVector Component', () => {
  beforeAll(() => server.listen());
  afterEach(() => {
    server.resetHandlers();
  });
  afterAll(() => server.close());

  const mockSetIsAddEditVectorVisible = jest.fn();
  const mockOnVectorUpdated = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  describe('Create Mode', () => {
    it('should render create vector modal with correct title', async () => {
      const initialState = createMockInitialState();
      renderWithRedux(
        <AddEditVector
          isAddEditVectorVisible={true}
          setIsAddEditVectorVisible={mockSetIsAddEditVectorVisible}
          vector={null}
          selectedGraph={mockGraphGUID}
        />,
        initialState,
        undefined,
        true
      );

      const modal = await screen.findByTestId('add-edit-vector-modal');
      expect(modal).toBeInTheDocument();
      expect(screen.getByText('Create Vector')).toBeInTheDocument();
    });

    it('should render all required form fields', async () => {
      const initialState = createMockInitialState();
      renderWithRedux(
        <AddEditVector
          isAddEditVectorVisible={true}
          setIsAddEditVectorVisible={mockSetIsAddEditVectorVisible}
          vector={null}
          selectedGraph={mockGraphGUID}
        />,
        initialState,
        undefined,
        true
      );

      await waitFor(
        () => {
          expect(screen.queryByText('Loading...')).not.toBeInTheDocument();
        },
        { timeout: 10000 }
      );

      // Check for form fields by placeholder text instead of labels
      expect(screen.getByPlaceholderText(/enter model/i)).toBeInTheDocument();
      expect(screen.getByPlaceholderText(/enter dimensionality/i)).toBeInTheDocument();
      expect(screen.getByPlaceholderText(/enter content/i)).toBeInTheDocument();

      // Check for JsonEditor - it appears as a textarea in the test environment
      expect(screen.getByTestId('json-editor-textarea')).toBeInTheDocument();

      // Check for Node and Edge selectors
      expect(screen.getByText('Node')).toBeInTheDocument();
      expect(screen.getByText('Edge')).toBeInTheDocument();
    });

    it('should have submit button disabled initially due to form validation', async () => {
      const initialState = createMockInitialState();
      renderWithRedux(
        <AddEditVector
          isAddEditVectorVisible={true}
          setIsAddEditVectorVisible={mockSetIsAddEditVectorVisible}
          vector={null}
          selectedGraph={mockGraphGUID}
        />,
        initialState,
        undefined,
        true
      );

      await waitFor(
        () => {
          expect(screen.queryByText('Loading...')).not.toBeInTheDocument();
        },
        { timeout: 10000 }
      );

      const submitButton = screen.getByRole('button', { name: /ok/i });
      expect(submitButton).toBeDisabled();
    });

    it('should show validation errors for empty required fields', async () => {
      const initialState = createMockInitialState();
      renderWithRedux(
        <AddEditVector
          isAddEditVectorVisible={true}
          setIsAddEditVectorVisible={mockSetIsAddEditVectorVisible}
          vector={null}
          selectedGraph={mockGraphGUID}
        />,
        initialState,
        undefined,
        true
      );

      await waitFor(
        () => {
          expect(screen.queryByText('Loading...')).not.toBeInTheDocument();
        },
        { timeout: 10000 }
      );

      // Try to submit without filling required fields
      const submitButton = screen.getByRole('button', { name: /ok/i });

      // The button should be disabled initially due to form validation
      expect(submitButton).toBeDisabled();

      // Fill only one field to trigger validation
      const modelInput = screen.getByPlaceholderText(/enter model/i);
      fireEvent.change(modelInput, { target: { value: 'test' } });

      // Try to submit again - should still be disabled or show validation errors
      fireEvent.click(submitButton);

      // Check if validation errors appear or button remains disabled
      await waitFor(() => {
        // Either validation errors should appear or button should remain disabled
        const isDisabled = submitButton.disabled;
        const hasValidationErrors = screen.queryByText(/please input/i);
        expect(isDisabled || hasValidationErrors).toBeTruthy();
      });
    });

    it('should close modal and reset form when cancel is clicked', async () => {
      const initialState = createMockInitialState();
      renderWithRedux(
        <AddEditVector
          isAddEditVectorVisible={true}
          setIsAddEditVectorVisible={mockSetIsAddEditVectorVisible}
          vector={null}
          selectedGraph={mockGraphGUID}
        />,
        initialState,
        undefined,
        true
      );

      await waitFor(
        () => {
          expect(screen.queryByText('Loading...')).not.toBeInTheDocument();
        },
        { timeout: 10000 }
      );

      const cancelButton = screen.getByRole('button', { name: /cancel/i });
      fireEvent.click(cancelButton);

      expect(mockSetIsAddEditVectorVisible).toHaveBeenCalledWith(false);
    });
  });

  describe('Edit Mode', () => {
    const mockVector: VectorType = {
      ...mockVectorData[0],
      NodeName: 'Test Node',
      EdgeName: 'Test Edge',
      LastUpdateUtc: mockVectorData[0].LastUpdateUtc || '2024-01-01T00:00:00Z',
    };

    it('should render edit vector modal with correct title', async () => {
      const initialState = createMockInitialState();
      renderWithRedux(
        <AddEditVector
          isAddEditVectorVisible={true}
          setIsAddEditVectorVisible={mockSetIsAddEditVectorVisible}
          vector={mockVector}
          selectedGraph={mockGraphGUID}
        />,
        initialState,
        undefined,
        true
      );

      const modal = await screen.findByTestId('add-edit-vector-modal');
      expect(modal).toBeInTheDocument();
      expect(screen.getByText('Edit Vector')).toBeInTheDocument();
    });

    it('should pre-populate form fields with vector data', async () => {
      const initialState = createMockInitialState();
      renderWithRedux(
        <AddEditVector
          isAddEditVectorVisible={true}
          setIsAddEditVectorVisible={mockSetIsAddEditVectorVisible}
          vector={mockVector}
          selectedGraph={mockGraphGUID}
        />,
        initialState,
        undefined,
        true
      );

      await waitFor(
        () => {
          expect(screen.queryByText('Loading...')).not.toBeInTheDocument();
        },
        { timeout: 10000 }
      );

      const modelInput = screen.getByPlaceholderText(/enter model/i);
      const dimensionalityInput = screen.getByPlaceholderText(/enter dimensionality/i);
      const contentInput = screen.getByPlaceholderText(/enter content/i);

      expect(modelInput).toHaveValue(mockVector.Model);
      expect(dimensionalityInput).toHaveValue(mockVector.Dimensionality);
      expect(contentInput).toHaveValue(mockVector.Content);
    });

    it('should successfully update vector with modified data', async () => {
      const initialState = createMockInitialState();
      renderWithRedux(
        <AddEditVector
          isAddEditVectorVisible={true}
          setIsAddEditVectorVisible={mockSetIsAddEditVectorVisible}
          vector={mockVector}
          selectedGraph={mockGraphGUID}
          onVectorUpdated={mockOnVectorUpdated}
        />,
        initialState,
        undefined,
        true
      );

      await waitFor(
        () => {
          expect(screen.queryByText('Loading...')).not.toBeInTheDocument();
        },
        { timeout: 10000 }
      );

      const modelInput = screen.getByPlaceholderText(/enter model/i);
      fireEvent.change(modelInput, { target: { value: 'updated-model' } });

      // Wait for form validation to complete
      await waitFor(
        () => {
          const submitButton = screen.getByRole('button', { name: /ok/i });
          expect(submitButton).not.toBeDisabled();
        },
        { timeout: 10000 }
      );

      const submitButton = screen.getByRole('button', { name: /ok/i });
      fireEvent.click(submitButton);

      await waitFor(
        () => {
          expect(mockSetIsAddEditVectorVisible).toHaveBeenCalledWith(false);
          expect(mockOnVectorUpdated).toHaveBeenCalled();
        },
        { timeout: 5000 }
      );
    });
  });

  describe('Update Vector Logic', () => {
    it('should handle undefined NodeGUID and EdgeGUID during update', async () => {
      const mockVector: VectorType = {
        ...mockVectorData[0],
        NodeName: 'Test Node',
        EdgeName: 'Test Edge',
        LastUpdateUtc: mockVectorData[0].LastUpdateUtc || '2024-01-01T00:00:00Z',
      };

      const initialState = createMockInitialState();
      renderWithRedux(
        <AddEditVector
          isAddEditVectorVisible={true}
          setIsAddEditVectorVisible={mockSetIsAddEditVectorVisible}
          vector={mockVector}
          selectedGraph={mockGraphGUID}
          onVectorUpdated={mockOnVectorUpdated}
        />,
        initialState,
        undefined,
        true
      );

      await waitFor(
        () => {
          expect(screen.queryByText('Loading...')).not.toBeInTheDocument();
        },
        { timeout: 10000 }
      );

      const modelInput = screen.getByPlaceholderText(/enter model/i);
      fireEvent.change(modelInput, { target: { value: 'updated-model' } });

      await waitFor(
        () => {
          const submitButton = screen.getByRole('button', { name: /ok/i });
          expect(submitButton).not.toBeDisabled();
        },
        { timeout: 10000 }
      );

      const submitButton = screen.getByRole('button', { name: /ok/i });
      fireEvent.click(submitButton);

      await waitFor(
        () => {
          expect(mockSetIsAddEditVectorVisible).toHaveBeenCalledWith(false);
          expect(mockOnVectorUpdated).toHaveBeenCalled();
        },
        { timeout: 5000 }
      );
    });
  });

  describe('Success Handling', () => {
    it('should show success toast and call onVectorUpdated after successful update', async () => {
      const mockVector: VectorType = {
        ...mockVectorData[0],
        LastUpdateUtc: mockVectorData[0].LastUpdateUtc || '2024-01-01T00:00:00Z',
      };

      const initialState = createMockInitialState();
      renderWithRedux(
        <AddEditVector
          isAddEditVectorVisible={true}
          setIsAddEditVectorVisible={mockSetIsAddEditVectorVisible}
          vector={mockVector}
          selectedGraph={mockGraphGUID}
          onVectorUpdated={mockOnVectorUpdated}
        />,
        initialState,
        undefined,
        true
      );

      await waitFor(
        () => {
          expect(screen.queryByText('Loading...')).not.toBeInTheDocument();
        },
        { timeout: 10000 }
      );

      const modelInput = screen.getByPlaceholderText(/enter model/i);
      fireEvent.change(modelInput, { target: { value: 'updated-model' } });

      await waitFor(
        () => {
          const submitButton = screen.getByRole('button', { name: /ok/i });
          expect(submitButton).not.toBeDisabled();
        },
        { timeout: 10000 }
      );

      const submitButton = screen.getByRole('button', { name: /ok/i });
      fireEvent.click(submitButton);

      await waitFor(
        () => {
          // Check that the toast.success was called instead of looking for the text in DOM
          const { success } = require('react-hot-toast');
          expect(success).toHaveBeenCalledWith('Vector updated successfully');
          expect(mockSetIsAddEditVectorVisible).toHaveBeenCalledWith(false);
          expect(mockOnVectorUpdated).toHaveBeenCalled();
        },
        { timeout: 5000 }
      );
    });
  });

  describe('Edge Cases', () => {
    it('should handle component unmounting during API call', async () => {
      const initialState = createMockInitialState();
      const { unmount } = renderWithRedux(
        <AddEditVector
          isAddEditVectorVisible={true}
          setIsAddEditVectorVisible={mockSetIsAddEditVectorVisible}
          vector={null}
          selectedGraph={mockGraphGUID}
        />,
        initialState,
        undefined,
        true
      );

      await waitFor(
        () => {
          expect(screen.queryByText('Loading...')).not.toBeInTheDocument();
        },
        { timeout: 10000 }
      );

      const modelInput = screen.getByPlaceholderText(/enter model/i);
      const dimensionalityInput = screen.getByPlaceholderText(/enter dimensionality/i);
      const contentInput = screen.getByPlaceholderText(/enter content/i);

      fireEvent.change(modelInput, { target: { value: 'test-model' } });
      fireEvent.change(dimensionalityInput, { target: { value: '3' } });
      fireEvent.change(contentInput, { target: { value: 'test content' } });

      const submitButton = screen.getByRole('button', { name: /ok/i });
      fireEvent.click(submitButton);

      // Unmount before API call completes
      unmount();

      // Should not throw any errors
      expect(true).toBe(true);
    });

    it('should handle rapid form changes', async () => {
      const initialState = createMockInitialState();
      renderWithRedux(
        <AddEditVector
          isAddEditVectorVisible={true}
          setIsAddEditVectorVisible={mockSetIsAddEditVectorVisible}
          vector={null}
          selectedGraph={mockGraphGUID}
        />,
        initialState,
        undefined,
        true
      );

      await waitFor(
        () => {
          expect(screen.queryByText('Loading...')).not.toBeInTheDocument();
        },
        { timeout: 10000 }
      );

      const modelInput = screen.getByPlaceholderText(/enter model/i);
      const dimensionalityInput = screen.getByPlaceholderText(/enter dimensionality/i);
      const contentInput = screen.getByPlaceholderText(/enter content/i);

      // Rapid changes
      fireEvent.change(modelInput, { target: { value: 'test1' } });
      fireEvent.change(modelInput, { target: { value: 'test2' } });
      fireEvent.change(modelInput, { target: { value: 'test3' } });
      fireEvent.change(dimensionalityInput, { target: { value: '1' } });
      fireEvent.change(dimensionalityInput, { target: { value: '2' } });
      fireEvent.change(dimensionalityInput, { target: { value: '3' } });
      fireEvent.change(contentInput, { target: { value: 'content1' } });
      fireEvent.change(contentInput, { target: { value: 'content2' } });
      fireEvent.change(contentInput, { target: { value: 'content3' } });

      await waitFor(() => {
        expect(modelInput).toHaveValue('test3');
        expect(dimensionalityInput).toHaveValue(3);
        expect(contentInput).toHaveValue('content3');
      });
    });

    it('should handle empty selectedGraph prop', async () => {
      const initialState = createMockInitialState();
      renderWithRedux(
        <AddEditVector
          isAddEditVectorVisible={true}
          setIsAddEditVectorVisible={mockSetIsAddEditVectorVisible}
          vector={null}
          selectedGraph=""
        />,
        initialState,
        undefined,
        true
      );

      await waitFor(
        () => {
          expect(screen.queryByText('Loading...')).not.toBeInTheDocument();
        },
        { timeout: 10000 }
      );

      const modal = screen.getByTestId('add-edit-vector-modal');
      expect(modal).toBeInTheDocument();
    });
  });
});
