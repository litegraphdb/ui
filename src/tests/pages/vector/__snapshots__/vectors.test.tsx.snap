// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vectors Page renders the vectors page: initial table state 1`] = `
<div>
  <main
    class="ant-layout-content pageContainer css-dev-only-do-not-override-98ntnt"
    id="vectors"
  >
    <div
      class="pageTitleContainer ant-flex css-dev-only-do-not-override-98ntnt ant-flex-wrap-wrap ant-flex-align-center ant-flex-justify-space-between ant-flex-gap-small"
    >
      <span
        class="ant-typography css-dev-only-do-not-override-98ntnt"
        data-testid="heading"
        style="font-weight: 600; font-size: 16px;"
      >
        <div
          class="ant-flex css-dev-only-do-not-override-98ntnt ant-flex-align-center"
          style="gap: 10px;"
        >
          <span
            class="ant-typography css-dev-only-do-not-override-98ntnt"
          >
            Vectors
          </span>
          <span
            aria-label="loading"
            class="anticon anticon-loading anticon-spin loading-icon"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="loading"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="0 0 1024 1024"
              width="1em"
            >
              <path
                d="M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 00-94.3-139.9 437.71 437.71 0 00-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3.1 19.9-16 36-35.9 36z"
              />
            </svg>
          </span>
        </div>
      </span>
      <button
        class="ant-btn css-dev-only-do-not-override-98ntnt ant-btn-link"
        style="font-weight: 500;"
        type="button"
      >
        <span
          class="ant-btn-icon"
        >
          <span
            aria-label="plus-square"
            class="anticon anticon-plus-square"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="plus-square"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M328 544h152v152c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V544h152c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8H544V328c0-4.4-3.6-8-8-8h-48c-4.4 0-8 3.6-8 8v152H328c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8z"
              />
              <path
                d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zm-40 728H184V184h656v656z"
              />
            </svg>
          </span>
        </span>
        <span>
          Create Vector
        </span>
      </button>
    </div>
    <div
      class="pageContent"
    >
      <div
        class="ant-table-wrapper css-dev-only-do-not-override-98ntnt"
      >
        <div
          class="ant-spin-nested-loading css-dev-only-do-not-override-98ntnt"
        >
          <div>
            <div
              aria-busy="true"
              aria-live="polite"
              class="ant-spin ant-spin-spinning css-dev-only-do-not-override-98ntnt"
            >
              <span
                class="ant-spin-dot-holder"
              >
                <span
                  class="ant-spin-dot ant-spin-dot-spin"
                >
                  <i
                    class="ant-spin-dot-item"
                  />
                  <i
                    class="ant-spin-dot-item"
                  />
                  <i
                    class="ant-spin-dot-item"
                  />
                  <i
                    class="ant-spin-dot-item"
                  />
                </span>
              </span>
            </div>
          </div>
          <div
            class="ant-spin-container ant-spin-blur"
          >
            <div
              class="ant-table ant-table-empty css-dev-only-do-not-override-98ntnt"
            >
              <div
                class="ant-table-container"
              >
                <div
                  class="ant-table-content"
                >
                  <table
                    style="table-layout: auto;"
                  >
                    <colgroup />
                    <thead
                      class="ant-table-thead"
                    >
                      <tr>
                        <th
                          class="ant-table-cell"
                          scope="col"
                        >
                          Actions
                        </th>
                      </tr>
                    </thead>
                    <tbody
                      class="ant-table-tbody"
                    >
                      <tr
                        class="ant-table-placeholder"
                      >
                        <td
                          class="ant-table-cell"
                        >
                          <div
                            class="css-dev-only-do-not-override-98ntnt ant-empty ant-empty-normal"
                          >
                            <div
                              class="ant-empty-image"
                            >
                              <svg
                                height="41"
                                viewBox="0 0 64 41"
                                width="64"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <title>
                                  Simple Empty
                                </title>
                                <g
                                  fill="none"
                                  fill-rule="evenodd"
                                  transform="translate(0 1)"
                                >
                                  <ellipse
                                    cx="32"
                                    cy="33"
                                    fill="#f5f5f5"
                                    rx="32"
                                    ry="7"
                                  />
                                  <g
                                    fill-rule="nonzero"
                                    stroke="#d9d9d9"
                                  >
                                    <path
                                      d="M55 12.76L44.854 1.258C44.367.474 43.656 0 42.907 0H21.093c-.749 0-1.46.474-1.947 1.257L9 12.761V22h46v-9.24z"
                                    />
                                    <path
                                      d="M41.613 15.931c0-1.605.994-2.93 2.227-2.931H55v18.137C55 33.26 53.68 35 52.05 35h-40.1C10.32 35 9 33.259 9 31.137V13h11.16c1.233 0 2.227 1.323 2.227 2.928v.022c0 1.605 1.005 2.901 2.237 2.901h14.752c1.232 0 2.237-1.308 2.237-2.913v-.007z"
                                      fill="#fafafa"
                                    />
                                  </g>
                                </g>
                              </svg>
                            </div>
                            <div
                              class="ant-empty-description"
                            >
                              No data
                            </div>
                          </div>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </main>
</div>
`;

exports[`Vectors Page should create a vector and should be visible in the table 1`] = `
<button
  class="ant-btn css-dev-only-do-not-override-98ntnt ant-btn-link"
  style="font-weight: 500;"
  type="button"
>
  <span
    class="ant-btn-icon"
  >
    <span
      aria-label="plus-square"
      class="anticon anticon-plus-square"
      role="img"
    >
      <svg
        aria-hidden="true"
        data-icon="plus-square"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M328 544h152v152c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V544h152c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8H544V328c0-4.4-3.6-8-8-8h-48c-4.4 0-8 3.6-8 8v152H328c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8z"
        />
        <path
          d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zm-40 728H184V184h656v656z"
        />
      </svg>
    </span>
  </span>
  <span>
    Create Vector
  </span>
</button>
`;

exports[`Vectors Page should create a vector and should be visible in the table: final table state 1`] = `null`;

exports[`Vectors Page should delete vector successfully: final table state 1`] = `null`;

exports[`Vectors Page should update vector successfully: final table state 1`] = `null`;
